src/runtime_src/core/pcie/tools/xbmgmt/xqspips.cpp

```diff
@@ -375,6 +376,37 @@ bool XQSPIPS_Flasher::waitTxEmpty()
     return false;
 }
 
+int XQSPIPS_Flasher::revertToMFG(void)
+{
+    initQSpiPS();
+
+    uint32_t StatusReg = XQSpiPS_GetStatusReg();
+
+    if (StatusReg == 0xFFFFFFFF) {
+        std::cout << "[ERROR]: Read PCIe device return -1. Cannot get QSPI status." << std::endl;
+        exit(-EOPNOTSUPP);
+    }
+
+    /* Make sure it is ready to receive commands. */
+    resetQSpiPS();
+    XQSpiPS_Enable_GQSPI();
+
+    if (!getFlashID()) {
+        std::cout << "[ERROR]: Could not get Flash ID" << std::endl;
+        exit(-EOPNOTSUPP);
+    }
+
+    /* Use 4 bytes address mode */
+    enterOrExitFourBytesMode(ENTER_4B);
+
+    // Sectoer size is defined by SECTOR_SIZE
+    std::cout << "Factory resetting " << std::flush;
+    eraseSector(0, GOLDEN_BASE - 3 * SECTOR_SIZE);
```

Comment:
The eraseSector() would erase extra sectors. It is okay when there is only one image. We would better roundup the byteCount to nearest SECTOR_SIZE in eraseSector().

Reply:
yes, eraseSector so far erases more than required. This will have issues when we have bin file ~96M size and we also try to hold gloden at 96M.
will fix eraseSector() to erase accurate number of sectors.